crates/core_simd/src/ops.rs

```diff
@@ -144,8 +144,15 @@ macro_rules! impl_op {
             impl<const LANES: usize> core::ops::Neg for crate::$type<LANES> {
                 type Output = Self;
                 fn neg(self) -> Self::Output {
-                    Self::splat(0) - self
-                }
+                // TODO probs a smarter way to do this too
+                    if AsRef::<[$scalar]>::as_ref(&self)
```

Comment:
I think you should be able to use `self.as_slice()` or `self.to_array()` here instead of UFCS.  This is fine but just a bit more verbose.

Reply:
Noted and changed.